pkg/ca/intermediateca/intermediateca.go

```diff
@@ -37,6 +42,88 @@ type IntermediateCA struct {
 	Signer crypto.Signer
 }
 
+func (ica *IntermediateCA) CreatePrecertificate(ctx context.Context, challenge *challenges.ChallengeResult) (*ca.CodeSigningPreCertificate, error) {
+	cert, err := x509ca.MakeX509(challenge)
+	if err != nil {
+		return nil, err
+	}
+
+	certChain, privateKey := ica.getX509KeyPair()
+
+	// Append poison extension
+	cert.ExtraExtensions = append(cert.ExtraExtensions, pkix.Extension{
+		Id:       asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 11129, 2, 4, 3},
```

Comment:
can you use https://github.com/google/certificate-transparency-go/blob/58784bdeda2ec153f22c23ff1d6cfcc6f1c7f5a2/x509/x509.go#L2234 instead of putting the ints inline here?

Reply:
Unforunately ct-go forked asn1 and x509 packages, so this `asn1` package does not match CT's `asn1` package. We could switch over to using the CT version of x509 and asn1, but it seems simpler to just copy the OIDs over.

Reply:
Moving the OIDs into constants.